### PR TITLE
Set root-path from uvicorn

### PR DESCRIPTION
**Pull request by mmcfarland**
## Description
Sets the root path in uvicorn to match that set via FastAPI. This is required with updates to starlette, which were included previously.

Fixes https://github.com/microsoft/PlanetaryComputer/issues/360

Also updates the tests to more closely match the FastAPI setup (including the use of root_path) as well as ensures that the dev dependencies are equivalent to those in the running container.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ services:
       - azurite
       - redis
     command: >
-      bash -c "pypgstac pgready && uvicorn pcstac.main:app --host 0.0.0.0 --port 8081 --reload --proxy-headers"
+      bash -c "pypgstac pgready && uvicorn pcstac.main:app --host 0.0.0.0 --port 8081 --reload --proxy-headers --root-path '/stac'"
```
Comment (mmcfarland) [inline docker-compose.yml:19]: Locally, this is the actual fix to the issue.

File: pcstac/Dockerfile [hunk 2]
```diff
@@ -22,4 +22,8 @@ RUN --mount=type=cache,target=/root/.cache \
 ENV APP_HOST=0.0.0.0
 ENV APP_PORT=81
 
-CMD uvicorn pcstac.main:app --host ${APP_HOST} --port ${APP_PORT} --log-level info
+# This value should match that which is used as the root_path in FastAPI, which
+# is typically set via the APP_ROOT_PATH environment variable.
+ENV APP_ROOT_PATH=""
+
+CMD uvicorn pcstac.main:app --host ${APP_HOST} --port ${APP_PORT} --root-path ${APP_ROOT_PATH} --log-level info
```
Comment (mmcfarland) [inline pcstac/Dockerfile:29]: Similarly, this fixes the issue in the deployed environment. `APP_ROOT_PATH` is already set in the Helm chart.

File: pcstac/Dockerfile.dev [hunk 3]
```diff
@@ -1,9 +1,8 @@
 FROM pc-apis-stac
 
-
 RUN --mount=type=cache,target=/root/.cache \
     --mount=type=bind,source=requirements-dev.txt,target=requirements-dev.txt \
     pip install -r requirements-dev.txt
 
 RUN --mount=type=cache,target=/root/.cache \
-    pip install -e ./pccommon[dev] -e ./pcstac
+    pip install --no-deps -e ./pccommon[dev] -e ./pcstac
```
Comment (mmcfarland) [inline pcstac/Dockerfile.dev:8]: Ensures that pcstac dependencies aren't overwritten by pccommon

File: pcstac/Dockerfile (hunk 2 above)
Comment (TomAugspurger) [reply pcstac/Dockerfile:29]: Just confirming the behavior here: The "default" `APP_ROOT_PATH` is set to `""`, but if the Helm chart defines it it'll override that?

And is the default of `""` is correct? Not `"/"`?

**Review by TomAugspurger — APPROVED** [review]

File: pcstac/Dockerfile (hunk 2 above)
Comment (mmcfarland) [reply pcstac/Dockerfile:29]: In practice, I didn't want to give it a default value. Locally and on PCT, it's `/stac` and in PC its `/api/stac/v1`.  The value is always set by either docker-compose or helm. I believe you need to declare the ENV in order for it be available when the container is run.

**Pull request merged**